service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -61,7 +61,7 @@
   private final OperationHandle opHandle;
   public static final FetchOrientation DEFAULT_FETCH_ORIENTATION = FetchOrientation.FETCH_NEXT;
   public static final Logger LOG = LoggerFactory.getLogger(Operation.class.getName());
-  protected boolean hasResultSet;
+  protected Boolean hasResultSet = null;
```

Comment:
Can revert back to be primitive boolean itself as we always return false if it is null.

Reply:
it is required for constructing OperationStatus